<launch>
<!-- Arguments -->
  <arg name="robot" default="nusim" />
  <arg name="cmd_src" default="teleop"/>
  <arg name="wheel_left" default="wheel_left_joint"/>
  <arg name="wheel_right" default="wheel_right_joint"/>
  <arg if="$(eval '\'$(var robot)\' == \'nusim\'')" name="slam" default="nusim" />
  <arg if="$(eval '\'$(var robot)\' != \'nusim\'')" name="slam" default="real" />

  <node pkg="tf2_ros" exec="static_transform_publisher" args="--frame-id nusim/world --child-frame-id map"/>
  <node pkg="tf2_ros" exec="static_transform_publisher" args="--frame-id map --child-frame-id green/odom"/>


<!--If robot = nusim, modify nusim rviz config with all 3 robots
    then call nusim launch passing in that config
    ros2 launch nuturtle_control start_robot.launch robot:=nusim-->
<!-- Start teleop node through this launch-->
  <include if="$(eval '\'$(var robot)\' == \'nusim\'')" file="$(find-pkg-share nuturtle_control)/launch/start_robot.launch">
        <arg name="robot" value="$(var robot)" />
        <arg name="cmd_src" value="$(var cmd_src)"/>
        <arg name="slam" value="$(var slam)"/>
  </include>

<!-- Start slam node -->
  <node pkg="nuslam" exec="slam" name="slam_node">
        <param name="body_id" value = "green/base_footprint"/>
        <param name="odom_id" value = "green/odom"/>
        <param name="wheel_left" value = "$(var wheel_left)"/>
        <param name="wheel_right" value = "$(var wheel_right)"/>
        <param from="$(find-pkg-share nuturtle_description)/config/diff_params.yaml"/>
  </node>

<!-- launch green boy-->
  <include file="$(find-pkg-share nuturtle_description)/launch/load_one.launch.py">
        <arg name="color" value="green" />
        <arg name="use_rviz" value="false" />
  </include> 

<!--If robot=ROBOTNAM -> impliment this next hw-->
</launch>


